src/postamble_minimal.js

```diff
@@ -48,6 +48,11 @@ function initRuntime(asm) {
 
 var imports = {
   'env': asmLibraryArg
+#if FILESYSTEM
+  // TODO: Fix size bloat coming from WASI properly. The -s FILESYSTEM=1 check is too weak to properly DCE WASI linkage away.
+  // (Emscripten now unconditionally uses WASI for stdio, perhaps replace that with web-friendly stdio)
+  , '{{{ WASI_MODULE_NAME }}}': asmLibraryArg
```

Comment:
Why is adding this line unconditionally a problem?   Its the same asmLibraryArg so it should be very minimcal increase.

Also, the WASI we use for stdio doesn't actually add any overhead IIRC.  We just renamed the existing syscalls.  We delibertely only converted the WASI the syscalls that didn't come with any code bloat (other than this one line to export them under a different module).

Reply:
Very minimal increase means a "> 0 increase". MINIMAL_RUNTIME does not accept such regressions.

Reply:
I think we should strongly consider this one though.  Sharing the WASI API for basic syscallsat the cost of just this one line is worth it IMHO.   I also believe that dividing the imports up into different modules is a nice for comprehension and debug-ability.    I was even thinking of going further than this and putting, for example, all the syscalls under `syscall` rather then `env`.

It could actually be size win too (although I guess minimize import/export names might nullify this) since:

```
`wasi` : {'func1': func1 .... funcN } 
```

can be smaller than

```
`env` : {'wasi_func1': func1 .... funcN } 
```

Reply:
Sure, but that is not what this is about. This is about doing a build that does not use WASI anywhere that still has to pay code lines regarding WASI. If a MINIMAL_RUNTIME-based build wants to target WASI and pay the code size, that would be fine, but for code that is not using WASI, requiring increased code size is not acceptable for minimal runtime.

In regular runtime things can be as lax as is convenient for any purpose, but MINIMAL_RUNTIME is the surface area where every single byte counts.

Reply:
At what optimization level does every byte count, in your opinion? If just in fully optimized builds, thenI think we can (and should) make import name minification also merge the separate import modules into a single one (also it could have the name "a" which would save 2 bytes compared to now, btw). I can look into that.

(For the immediate PR, as mentioned earlier, perhaps the test could be modified, since it's about memory anyhow and not printing?)

Reply:
> At what optimization level does every byte count, in your opinion? If just in fully optimized builds

Yes, in fully optimized builds.

> we can (and should) make import name minification also merge the separate import modules into a single one

That would be pretty cool!